### PR TITLE
Adding MDCSpanDecorator

### PR DESCRIPTION
**Pull request by ledor473**
This PR implements an MDCSpanDecorator and a HeaderSpanDecorator to decorate incoming span on the servlet-filter.

This has been in use internally for some time and we've found that both Header and MDC are useful when adding tracing to existing service.

There will be other related PR to `java-spring-web` and `java-spring-cloud`

cc @jfbreault

### CODE REVIEW

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/mdc/MDCSpanDecorator.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package io.opentracing.contrib.mdc;
+
+import io.opentracing.Span;
+import io.opentracing.tag.StringTag;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.MDC;
+
+public class MDCSpanDecorator {
```
Comment (pavolloffay) [inline opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/mdc/MDCSpanDecorator.java]: How is this loaded?

File: opentracing-web-servlet-filter/pom.xml [hunk 2]
```diff
@@ -27,10 +27,16 @@
...
     <version.com.squareup.okhttp3>3.5.0</version.com.squareup.okhttp3>
+    <version.org.mockito>1.10.19</version.org.mockito>
+    <version.ch.qos.logback>1.1.11</version.ch.qos.logback>
     <main.basedir>${project.basedir}/..</main.basedir>
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
```
Comment (pavolloffay) [inline opentracing-web-servlet-filter/pom.xml]: I am not sure if we want to include even this. There should be perhaps a separate MDC instrumentation

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/decorator/ServletFilterHeaderSpanDecorator.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+import io.opentracing.Span;
+import io.opentracing.contrib.web.servlet.filter.ServletFilterSpanDecorator;
+import io.opentracing.tag.StringTag;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ServletFilterHeaderSpanDecorator implements ServletFilterSpanDecorator {
```
Comment (pavolloffay) [inline opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/decorator/ServletFilterHeaderSpanDecorator.java:11]: This could be in a separate PR

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/mdc/MDCSpanDecorator.java (hunk 1 above)
Comment (ledor473) [reply opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/mdc/MDCSpanDecorator.java]: This library don't load any Decorator, it's left to the user to do so **or** to the SpringBoot library (PR is opened for that as well)

File: opentracing-web-servlet-filter/pom.xml (hunk 2 above)
Comment (ledor473) [reply opentracing-web-servlet-filter/pom.xml]: Yeah I wasn't sure neither. But felt that slf4j-api was pretty minimal. We could split it in aa different Maven module as well

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/decorator/ServletFilterHeaderSpanDecorator.java (hunk 3 above)
Comment (ledor473) [reply opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/decorator/ServletFilterHeaderSpanDecorator.java:11]: If we decide to split MDC and Headers, yes

**Comment by ledor473** [comment]
@pavolloffay would you prefer that I create a new Maven module to host the MDC-specific code?
It's not much to do for me, but I believe you will have to adjust a few things in Bintray

**Comment by pavolloffay** [comment]
Do you mean a submodule in this project? To do that we don't have to change anything in bintray.

**Comment by ledor473** [comment]
So before I start changing things around, would creating a `java-web-servlet-filter-mdc` be enough?

**Comment by ledor473** [comment]
@pavolloffay I've split the MDC in a separate submodule in this project. Let me know if there's anything outstanding that prevent this to be merged 👍

**Comment by ledor473** [comment]
@pavolloffay I've split this pull request in two PRs and rebase them accordingly.

We would need to merge this first: https://github.com/opentracing-contrib/java-web-servlet-filter/pull/51

When https://github.com/opentracing-contrib/java-web-servlet-filter/pull/51 is merged, this one will only show the MDC code.

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/decorator/ServletFilterHeaderSpanDecorator.java (hunk 3 above)
Comment (ledor473) [reply opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/decorator/ServletFilterHeaderSpanDecorator.java:11]: Done

**Comment by ledor473** [comment]
This PR will likely be moved under https://github.com/opentracing-contrib/java-api-extensions
